### ISSUE TITLE
Pentadiagonal solver

### ISSUE THREAD

**Issue by jpbrodrick89** [jpbrodrick89/jaxtra#7]
### Public API

Exposed at `jaxtra.lax.linalg.pentadiagonal_solve` (should mirror `jax.lax.linalg.tridiagonal_solve` using) implemented at `jaxtra._src.lax.linalg.pentadiagonal_solve` (again mirroring tridiagonal_solve). Diagonals should be called ds, dl, d, du and dw in line with Cuparse gpsv.

### Primitive (Private)

`jaxtra._src.lax.linalg.pentadiagonal_solve_p` (mirroring `jax._src.lax.linalg.tridiagonal_solve_p` lines 2523-2655) should have shape, jvp, transpose and batching rule.

### CPU Lowering

lapack's gbsv, look at how jax handles gtsv for reference and how scipy uses gbsv

### GPU Lowering

CuSparse's gpsvInterleavedBatch.  This is probably the harder bit as this is the is quite different to how gtsv is built into Jax. Jax uses vanilla gtsv and gtsvStridedBatch but cusparse ONLY has gpsvInterleavedBatch which may have a slightly different API/handling. I'm guess we might have to add an extra dimension to the inputs if using it not under vmap but not sure. Implementation should be in files such as csrc/gpu/jaxlib/gpu/sparse_kernels... (following how jax does gtsv). We will also need to add a file jaxtra._src.lib.gpu_sparse,py for registration. Consider carefully whether this counts as "_batch_partitionable".

I can provide you cuSparse docs if you don't have access.

### Pure Jax fallback (..._jax_impl)

I'm not sure if it's possible to do a stable, robust version of the Thomas algorithn im pure Jax or if we should just fallback to LU for simplicity.

### Chores

Add `register_module_custom_calls(gpu_sparse)` to `jaxtra._src.lax.linalg`.

### Testing

New file tests/test_pentadiagonal.py Compare against LU for a random pentadiagonal 10x10 matrix and also for a hyperdiffusion operator. Make sure you test without jit, with jit, with vmap and gradient. Look at jax's tridiagonal tests in jax/tests/linalg_test.py and jax/tests/linalg_sharding_test.py for inspiration.

### Benchmarking

Rename the current benchmarks "Least Squares", add a new section "Banded". Compare pentadiagonal to Cholesky for random symmetric positive definite pentadiagonal matrices for sizes us to 10k x 10k (follow existing benchmark patterns and guidelines).

@claude can you create a PR for this please make sure to following CONTRIBUTING.md, any questions feel free to comment in this issue or on your PR.
